lib/avromatic/model/validation.rb

```diff
@@ -0,0 +1,48 @@
+module Avromatic
+  module Model
+    module Validation
+      extend ActiveSupport::Concern
+
+      module ClassMethods
+
+        # Returns an array of messages
+        def validate_nested_value(value)
+          case value
+          when Avromatic::Model::Attributes
+            validate_record_value(value)
+          when Array
+            value.map.with_index do |element, index|
+              validate_nested_value(element).map do |message|
+                "[#{index}]#{message}"
```

Comment:
Full error messages roughly have the form: "#{field_name.humanize} #{message}", so a humanized top-level field name will be prepended to whatever is returned here. 

Some examples:
```
# top-level array "my_widgets", where widgets are records with a required name and price:
My widgets [1].name can't be blank

# Top-level hash, "all_widgets", by name with a missing price:
All widgets ['sprocket'].price can't be blank

# For a top-level "this_widget", a parent without a price:
This widget parent.price can't be blank

# For a top-level union "this_or_that":
This or that price can't be blank
```